flex-shrink: add description and detailed example (#34420)
mdn/content: files/en-us/web/css/flex-shrink/index.md

This property deals with situations where the browser calculates the flex-basis values of the flex items, and finds that they are too large to fit into the flex container. As long as flex-shrink has a positive value the items will shrink in order that they do not overflow the container.

The `flex-grow` deals with adding available space, and distributes available space proportional to each items flex grow factor, with the value of the `flex-grow` property the only consideration. The `flex-shrink` manages taking away space to make boxes fit into their container without overflowing, but the removal of space is a bit more complicated. The shrink factor is proportional based on how much an item is able to shrink. The flex shrink factor is multiplied by the flex base size when distributing negative space. This distributes negative space in proportion to how much the item is able to shrink. This prevents smaller items from shrinking to `0px` before a larger item is noticeably reduced.

> Generally, `flex-shrink` is used alongside the {{cssxref("flex-grow")}} and {{cssxref("flex-basis")}}. It is recommended to use the {{cssxref("flex")}} shorthand shorthand instead of separate `flex-grow`, `flex-shrink`, and `flex-basis` declarations. Within the `flex` shorthand, the shrink factor is always the second `<number>`. If the shorthand doesn't include two number values, `flex

Reviewer: ```suggestion
Generally, `flex-shrink` is used alongside the {{cssxref("flex-grow")}} and {{cssxref("flex-basis")}} properties. Within the `flex` shorthand, the shrink factor is always the second `<number>`. If the shorthand only includes one number value, that value is assumed to be the `flex-grow` value.
```

Reviewer: I'm assuming this is how you meant to end the sentence. I also removed a bit that you had repeated from earlier.

Author: yup. i ididn't notice that i had forgotten to finish my thought.

### Setting flex item shrink factor

> In this example, there are five flex items with a `flex-shrink` value greater than 0, that have combined width that is wider than their parent flex container, demonstrating how negative free space is distributed based on the item's shrink factor.

Reviewer: ```suggestion
This example demonstrates how negative free space is distributed based on the item's shrink factor. It includes five flex items with a `flex-shrink` value greater than 0, which have a combined width greater than their parent flex container's width.
```

#### HTML

```html
<div id="content">
  <div class="box" style="background-color:red;">A</div>
  <div class="box" style="background-color:lightblue;">B</div>
  <div class="box" style="background-color:yellow;">C</div>
  <div class="box4" style="background-color:lightsalmon;">D</div>
  <div class="box5" style="background-color:lightgreen;">E</div>
</div>
```

#### CSS

> We give each flex item a {{cssxref("width")}} of `200px`. As the {{cssxref("flex-basis")}} property defaults to `auto`, each item's flex basis is `200px`, for a total of `1000px`; twice the size of the container. We set all flex items to be shrinkable, with `flex-shrink` values greater than `0`. The last two items have a greater `flex-shrink` values set, so they will shrink more.

Reviewer: ```suggestion
We give each flex item a {{cssxref("width")}} of `200px`. As the {{cssxref("flex-basis")}} property defaults to `auto`, each item's flex-basis is `200px`. This gives the flex items a total width of `1000px`, twice the size of the container. We set all flex items to be shrinkable, with `flex-shrink` values greater than `0`. The last two items have greater `flex-shrink` values set so they will shrink more.
```

```css
#content {
  display: flex;
  width: 500px;
}

#content div {
  width: 200px;
}

.box {
  flex-shrink: 1;
}

.box4 {
  flex-shrink: 1.5;
}

.box5 {
  flex-shrink: 2;
}
```

```css hidden
#content {
  margin: 5px;
}
div {
  font-family: monospace;
  outline: 1px solid;
  line-height: 4em;
  text-align: center;
}
```

#### Result

{{EmbedLiveSample('Setting_flex_item_shrink_factor', 500, 100)}}

> The `1000px` worth of flex items don't overflow their `500px` container because the items were able to shrink, distributing the `500px` of negative free space among the five items based on . The first three items have `flex-shrink: 1` set. D has `flex-shrink: 1.5` and E has `flex-shrink: 2` set. The final width of D and E is less than the others, with E smaller than D.

Reviewer: ```suggestion
The flex items don't overflow their container because they are able to shrink: the `500px` of negative free space is distributed among the five items based on their flex-shrink` values. The first three items have `flex-shrink: 1` set. D has `flex-shrink: 1.5` and E has `flex-shrink: 2` set. The final width of D and E is less than the others, with E smaller than D.
```
